packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -62,13 +63,23 @@ class PrecacheController {
 
     for (const entry of entries) {
       const {cacheKey, url} = createCacheKey(entry);
+
       if (this._urlsToCacheKeys.has(url) &&
           this._urlsToCacheKeys.get(url) !== cacheKey) {
         throw new WorkboxError('add-to-cache-list-conflicting-entries', {
           firstEntry: this._urlsToCacheKeys.get(url),
           secondEntry: cacheKey,
         });
       }
+
+      if (typeof entry !== 'string' && entry.integrity) {
+        this._urlsToIntegrity.set(cacheKey, entry.integrity);
+      } else {
+        // If we don't have an integrity value, then clear things out, just in
+        // case addToCacheList() was called multiple times: last call wins.
+        this._urlsToIntegrity.delete(cacheKey);
```

Comment:
We throw in the case of duplicate cache keys, why wouldn't we just do the same in the case of duplicate URLs with different integrity values? I don't see the value in support last call wins here.

Reply:
Okay, fair enough. I just went with that as my first instinct, but I'm happy to change.

Reply:
(I realize that one drawback is that this is going to add probably a couple of hundred extra bytes to the `PrecacheController` size for the error message and detailed check, all for a feature that is not going to be used by 99% of developers. Not sure if that should factor in to how we handle it.)

Reply:
Yeah, I think that's OK. In this case I don't think it's enough bytes to really worry about.